pgo/pgo.py

```diff
@@ -661,6 +661,27 @@ async def train_decommission(executable: PathLike, workdir: PathLike) -> None:
 trainers["decommission"] = ("decommission_dataset", train_decommission)
 populators["decommission_dataset"] = populate_decommission
 
+# AUTH CONNECTIONS STRESS ==================================================
+
+async def populate_auth_conns(executable: PathLike, workdir: PathLike) -> None:
+    # Create roles, table and permissions via CQL script.
+    async with with_cs_populate(executable=executable, workdir=workdir) as server:
+        await bash(fr"python3 ./exec_cql.py --file conf/auth.cql --host {server}")
+
+async def train_auth_conns(executable: PathLike, workdir: PathLike) -> None:
+    # Repeatedly connect as the reader user and perform simple reads to stress
+    # authentication, authorization and connection setup paths.
+    async with with_cluster(executable=executable, workdir=workdir) as (addrs, procs):
+        await asyncio.sleep(5)  # temporary gossip settle similar to other workloads
+        reader_user = "reader_role"
+        reader_pass = "password1"
+        # run 30k connection requests
+        await run_checked(["python3", "./auth_conns_stress.py", addrs[0], reader_user, reader_pass, "30", "10", "100"])
+    await merge_profraw(workdir)
```

Comment:
Normally I'd be very unhappy at forking 30k processes, but here I guess the goal is to simulate client startup, so it's sensible.

Reply:
I think we're forking 30 processes, with 10 threads each.

Reply:
I fork 30. Each is 10 threads (this is mostly done as I was fiddling with locally).

I will add named arguments to the script to make it more readable as Michal mentioned below.